cmd/main.go

```diff
@@ -300,9 +302,20 @@ func RunRootCommand(ctx context.Context, opt Options, args []string) error {
 	}
 
 	if opt.MCPServer {
-		if err = startMCPServer(ctx, opt); err != nil {
+		if err = StartMCPServer(ctx, opt); err != nil {
 			return fmt.Errorf("failed to start MCP server: %w", err)
 		}
+		return nil // MCP server mode blocks, so we return here
+	}
+
+	// Initialize MCP client if requested
+	if opt.MCPClient {
+		if err := tools.InitializeMCPClient(); err != nil {
+			klog.Warningf("Failed to initialize MCP client: %v", err)
+			// Continue with kubectl functionality - MCP failure should not block core features
```

Comment:
if mcp configuration exists, and we fail, then we should fail early to let the users know.